controllers/nodehealthcheck_controller.go

```diff
@@ -435,7 +435,10 @@ func (r *NodeHealthCheckReconciler) matchesUnhealthyConditions(nhc *remediationv
 			if now.After(n.LastTransitionTime.Add(c.Duration.Duration)) {
 				// unhealthy condition duration expired, node is unhealthy
 				r.Log.Info("Node matches unhealthy condition", "node", node.GetName(), "condition type", c.Type, "condition status", c.Status)
-				commonevents.NormalEventf(r.Recorder, nhc, utils.EventReasonDetectedUnhealthy, "Node matches unhealthy condition. Node %q, condition type %q, condition status %q", node.GetName(), c.Type, c.Status)
+				//In case a remediation is already created no need to repeat this event
```

Comment:
Calling `isRemediatingNode` inside the for loop increases the complexity of the function, but IIUC, we can call it before the loop and store the result to be used later.

Reply:
Hmm I don't think it increases the function complexity because nesting level stays the same either way (since the `if` block is still needed) + we need to introduce another local variable.

It might save a bit of time though because **isRemediatingNode** will be evaluated only once instead - not a real issue IMO but I don't feel strongly either way so I don't mind the change.

Reply:
If we move `isRemediating` outside the main loop the nesting level is not the same, or am I missing something?

Reply:
the way I see it it's:

```
if now.After(n.LastTransitionTime.Add(c.Duration.Duration)) {
	// unhealthy condition duration expired, node is unhealthy
	r.Log.Info("Node matches unhealthy condition", "node", node.GetName(), "condition type", c.Type, "condition status", c.Status)
	//In case a remediation is already created no need to repeat this event
	if !r.isRemediatingNode(nhc.Status.UnhealthyNodes, node.GetName()) {
		commonevents.NormalEventf(r.Recorder, nhc, utils.EventReasonDetectedUnhealthy, "Node matches unhealthy condition. Node %q, condition type %q, condition status %q", node.GetName(), c.Type, c.Status)
	}
	return true, nil
}
```
**Original**

Vs

```
if now.After(n.LastTransitionTime.Add(c.Duration.Duration)) {
	// unhealthy condition duration expired, node is unhealthy
	r.Log.Info("Node matches unhealthy condition", "node", node.GetName(), "condition type", c.Type, "condition status", c.Status)
	//In case a remediation is already created no need to repeat this event
	if !isNodeRemediating {
		commonevents.NormalEventf(r.Recorder, nhc, utils.EventReasonDetectedUnhealthy, "Node matches unhealthy condition. Node %q, condition type %q, condition status %q", node.GetName(), c.Type, c.Status)
	}
	return true, nil
}
```
**Modified**

Both are the same nesting level, because the `if` block still exist in both of them.
Maybe I've misunderstood your original idea ?